lib/palindrome_check.rb

```diff
@@ -1,5 +1,44 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  # check for nil
+  unless my_phrase.nil?
+    # remove all space from phrase
+    my_phrase = remove_spaces(my_phrase)
+    # call method to reverse the phrase
+     return check_pal(my_phrase)
+  end
+
+  return false
+    # raise NotImplementedError
 end
+
+def remove_spaces(phrase)
+  i = 0
+  n = phrase.length
+  no_space_phrase = ''
+  while i < n
+    unless phrase[i] == ' '
+      no_space_phrase += phrase[i]
+    end
+    i += 1
+  end
+
+  return no_space_phrase
+end
+
+def check_pal(phrase)
+  i = 0
+  last = phrase.length - 1
+  flag = true
+  while i < last
+    if phrase[i] != phrase[last]
+      flag = false
```

Comment:
Instead of setting the flag as false, you could consider stopping the loop and returning false here. If the first check itself does not match, you don't need to check the rest to conclude that it is not a palindrome.

Reply:
Ah right, good catch!